python/pyspark/sql/readwriter.py

```diff
@@ -441,7 +441,12 @@ def csv(self, path, schema=None, sep=None, encoding=None, quote=None, escape=Non
                   When it meets a record having fewer tokens than the length of the schema, \
                   sets ``null`` to extra fields. When the record has more tokens than the \
                   length of the schema, it drops extra tokens.
-                * ``DROPMALFORMED`` : ignores the whole corrupted records.
+                * ``DROPMALFORMED`` : ignores the whole corrupted records. Note that when CSV \
+                  parser column pruning (``spark.sql.csv.parser.columnPruning.enabled``) is enabled \
```

Comment:
One concern is that `spark.sql.csv.parser.columnPruning.enabled` is internal config. I'm not sure if there is policy that we shouldn't mention internal config in doc.

cc @HyukjinKwon @dongjoon-hyun